example/main.rs

```diff
@@ -0,0 +1,171 @@
+use dusk_bls12_381_sign::SecretKey;
+use rand::rngs::StdRng;
+use rand_core::SeedableRng;
+use std::thread;
+use std::time::Duration;
+use tokio::sync::mpsc;
+
+use consensus::commons::RoundUpdate;
+use consensus::consensus::Consensus;
+use consensus::messages::Message;
+use consensus::user::provisioners::{Provisioners, DUSK};
+
+use consensus::util::pending_queue::PendingQueue;
+use consensus::util::pubkey::PublicKey;
+use tokio::time;
+
+const MOCKED_PROVISIONERS_NUM: u64 = 5;
+
+fn generate_keys(n: u64) -> Vec<(SecretKey, PublicKey)> {
+    let mut keys = vec![];
+
+    for i in 0..n {
+        let rng = &mut StdRng::seed_from_u64(i);
+        let sk = dusk_bls12_381_sign::SecretKey::random(rng);
+        keys.push((
+            sk,
+            PublicKey::new(dusk_bls12_381_sign::PublicKey::from(&sk)),
+        ));
+    }
+
+    keys
+}
+
+fn generate_provisioners_from_keys(keys: Vec<(SecretKey, PublicKey)>) -> Provisioners {
+    let mut p = Provisioners::new();
+
+    for (pos, keys) in keys.into_iter().enumerate() {
+        p.add_member_with_value(keys.1, 1000 * (pos as u64) * DUSK);
+    }
```

Comment:
`enumerate` starts from 0, so I believe you're adding a provisioner with a 0 value

Furthermore, to increase the readability can be improved getting rid of `keys.1` this way
```suggestion
    for (pos, (_, pk)) in keys.into_iter().enumerate() {
        p.add_member_with_value(pk, 1000 * (pos as u64) * DUSK);
    }
```

Reply:
Thanks. Done.